### PR TITLE
`logs id` command

### PR DESCRIPTION
**Pull request by Cordila**
Added command `logs id <key>`, this helps users locate the log link / preview if they just have the log id.

This will help once a Heroku alternative is found and users log link changes due to moving from Heroku. They can use this command to bring up the new link easily

Have tested this.

Remade the PR because I changed the branch

### CODE REVIEW

File: cogs/modmail.py [hunk 1]
```diff
@@ -1212,6 +1212,30 @@ async def logs_closed_by(self, ctx, *, user: User = None):
...
+        logs = await self.bot.api.find_log_entry(key)
+
+        if not any(not log["open"] for log in logs):
+            embed = discord.Embed(
+                color=self.bot.error_color,
+                description=f"Log entry `{key}` not found.",
+            )
+            return await ctx.send(embed=embed)
+
+        logs = reversed(log for log in logs if not log["open"])
```
Comment (Taaku18) [inline cogs/modmail.py:1232]: This line triggers a TypeError: `TypeError: 'generator' object is not reversible`

**Review by Taaku18 — CHANGES_REQUESTED** [review]
Thanks for the PR. Summarizing from my comments: I prefer if you rename the command from logs id -> logs key, and remove the open check from L1225 to L1232.

File: cogs/modmail.py (hunk 1 above)
Comment (Cordila) [reply cogs/modmail.py:1232]: I did not face that error when I tested it, I followed the `logs` command to maintain consistency
https://github.com/modmail-dev/modmail/blob/1f3bd0e9fcecb0be1feb217f174b47bb1d5e2251/cogs/modmail.py#L1184

Not sure if I am missing something

File: cogs/modmail.py [hunk 2]
```diff
@@ -1212,6 +1212,30 @@ async def logs_closed_by(self, ctx, *, user: User = None):
...
+    @checks.has_permissions(PermissionLevel.SUPPORTER)
+    async def logs_id(self, ctx, key: str):
+        """
+        Get the log link for the specified log ID
+        """
+        icon_url = ctx.author.avatar.url
+
+        logs = await self.bot.api.find_log_entry(key)
+
+        if not any(not log["open"] for log in logs):
```
[the remark this answers is not in the document]
Comment (Cordila) [reply cogs/modmail.py:1225]: Was trying to maintain consistency with the `logs` command 
https://github.com/modmail-dev/modmail/blob/1f3bd0e9fcecb0be1feb217f174b47bb1d5e2251/cogs/modmail.py#L1177

[the remark this answers is not in the document]
Comment (Cordila) [reply cogs/modmail.py:1225]: My bad, was trying to maintain consistency but mistook the logs closed-by command

File: cogs/modmail.py (hunk 1 above)
Comment (Cordila) [reply cogs/modmail.py:1232]: It is throwing the error now for some reason yeah, lemme go have a look

File: cogs/modmail.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Cordila) [reply cogs/modmail.py:1225]: Was to maintain consistency with logs command https://github.com/Cordila/modmail/blob/53c65076545c03763870dc33b785c845dab288d9/cogs/modmail.py#L1177

[the remark this answers is not in the document]
Comment (khakers) [reply cogs/modmail.py:1225]: There's some possible rationales for that with the logs command but those don't apply to this one imo. Would definitely vote for not filtering by status.

[the remark this answers is not in the document]
Comment (Cordila) [reply cogs/modmail.py:1225]: I mean you dont even really get the log key until it is closed unless you use the `logs` command inside the thread to return the log link and then use that key to generate the log link again?

**Comment by Cordila** [comment]
Have completed the requested changes. 

A closed log will show this 
image

While if a log is open it will show this
image
I think this was the reason if logs were open we did not fetch the link. 

Can see here it is a commented known bug
https://github.com/modmail-dev/modmail/blob/d6eba12937d4e0024e1c9208d0ac22b365829397/cogs/modmail.py#L758

File: cogs/modmail.py [hunk 3]
```diff
@@ -1212,24 +1212,24 @@ async def logs_closed_by(self, ctx, *, user: User = None):
...
-        if not any(not log["open"] for log in logs):
+        if not logs:
             embed = discord.Embed(
                 color=self.bot.error_color,
                 description=f"Log entry `{key}` not found.",
             )
             return await ctx.send(embed=embed)
 
-        logs = reversed(log for log in logs if not log["open"])
+        logs = list(reversed(logs))
```
Comment (Taaku18) [inline cogs/modmail.py:1232]: There should only be at most one log with a certain key, there's really no use for the extra `list` and `reversed`. I'd suggest removing this line.

**Review by Taaku18 — CHANGES_REQUESTED** [review]
Everything else looks great, thanks for the changes.

**Review by Taaku18 — DISMISSED** [review]

**Pull request merged**